app/controllers/users_controller.rb

```diff
@@ -0,0 +1,57 @@
+class UsersController < ApplicationController
+
+  before_action :ensure_current_user, except: [:new, :create, :show]
```

Comment:
This is in a number of controllers - I'd put it in ApplicationController and then specify exceptions here with skip_before_action